projects/binutils/fuzz_bfd.c

```diff
@@ -1,28 +1,36 @@
+/* Copyright 2020 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
 #include "sysdep.h"
 #include "bfd.h"
 
 #include <stdint.h>
 #include <stdio.h>
+#include <unistd.h>
 
-
-static int bufferToFile(const char * name, const uint8_t *Data, size_t Size) {
-    FILE * fd;
-    if (remove(name) != 0) {
-        if (errno != ENOENT) {
-            printf("failed remove, errno=%d\n", errno);
-            return -1;
-        }
-    }
-    fd = fopen(name, "wb");
-    if (fd == NULL) {
-        printf("failed open, errno=%d\n", errno);
+static int bufferToFile(char * name, const uint8_t *Data, size_t Size) {
+    int fd = mkstemp(name);
```

Comment:
does this mean we will create a new temporary file for every execution of `LLVMFuzzerTestOneInput`?

Reply:
Yes it does.
Do you want me to remove it in the end of the fuzz target ?

Reply:
What's wrong with using the same file again and again? Especially, if we're willing to remove it after every iteration anyway?

Reply:
Concurrent runs I guess, cf https://github.com/google/oss-fuzz/issues/3506#issue-581154808

Reply:
I don't think we run fuzz targets in parallel, but to address this concern you could use PID in the filename

Reply:
@amodra, will just adding PID to the file name be enough for you ?
Or are you using a fuzz driver with multiple threads ?

> I don't think we run fuzz targets in parallel, but to address this concern you could use PID in the filename

@Dor1s ok but I think this is good to allow parallel fuzzing as oss-fuzz is not the only infrastructure running the fuzzers

Reply:
I'm not at all concerned about PID in filename.  In fact, that would be completely unnecessary even with multiple threads.  (Some implementations of mkstemp use the PID in XXXXXX as an initial try at a unique file name.)

I am concerned that you did not remove the abort.  That should go.  A bufferToFile fail of some sort should just result in LLVMFuzzerTestOneInput returning zero.  I don't think we want to open issues against bfd for fuzzer infrastructure fails.

Reply:
> I am concerned that you did not remove the abort. That should go

Ok Alan, now I get your point

> I don't think we want to open issues against bfd for fuzzer infrastructure fails.

I do think that we want some feedback when the fuzzer infrastructure fails.
@Dor1s what do you think ?

Reply:
- we can't accept a fuzz target that would leave a bunch of temp files after running, it should be either a finite number of small files, or all files must be deleted by the fuzzer
- abort() should be called if you want a bug to be reported. I can't say whether that condition is something you want to have a bug for or not -- totally up to you

Reply:
Ok, so the latest commit deletes the files, and do not use `abort` but `return 0` instead

Thanks :-)